### ISSUE TITLE
Error on attaching exceptions to spans

### ISSUE THREAD

**Issue by Wolg** [DataDog/dd-trace-php#693]
Hi, I got strange behaviour with `codeigniter` integration

```
Fatal error:Undefined variable: ex
/opt/datadog-php/dd-trace-sources/src/DDTrace/Integrations/CodeIgniter/V2/CodeIgniterSandboxedIntegration.php:150
```

It seems that in the same environment even simple checks fail when passing exception parameter, something like this:
```
function test() {
	echo 'test';
}
dd_trace_function("test", function ($s, $a, $r, $e) {
	var_dump($e === false);
});
test();
```
causes `Fatal error:Undefined variable: e`. If `test` function will throw an exception it'll be passed into the closure and `$e` variable will be defined.

Is this known? Please advice, thank you!

PHP 7.2.23 dd-trace `35.0`

**Comment by Wolg** [comment]
Ok, dived deep, so `Fatal error` is happening from my code 😞

**Comment by Wolg** [comment]
Ok, dived deep, so `Fatal error` is happening from my code 😞 
basically I have something like this there:
```
register_shutdown_function(function() {
    $error = error_get_last(); // Here comes Notice Undefined variable: e
    //$error through handling process gets promoted to 'Fatal Error'
});
```
From my pov, absence of exception shouldn't trigger notices, shall it be fixed on integrations or extension level?

**Comment by SammyK** [comment]
Thank you so much for the details @Wolg! You're absolutely right, errors that are muted from the tracer should not appear in custom error handlers as in your case. We're able to reproduce this issue on our end. We will update this thread when we have a fix on the way.

**Comment by morrisonlevi** [comment]
Thanks again for a detailed issue! The fix for this was included in [release 0.37.0](https://github.com/DataDog/dd-trace-php/releases/tag/0.37.0).

**Comment by Wolg** [comment]
Confirm, issue is fixed, thank you guys!
